freshmaker/pyxis.py

```diff
@@ -215,6 +215,121 @@ def get_images_by_digests(self, digests):
+                continue
+            tags = repo.get('auto_rebuild_tags')
+            # If the repository doesn't have 'auto_rebuild_tags', don't proceed with it
+            if tags:
+                reg_repo_info[reg_repo_pair]['auto_rebuild_tags'] = set(tags)
+            else:
+                del reg_repo_info[reg_repo_pair]
+
+    def _filter_auto_rebuild_nvrs(self, reg_repo_info):
+        """
+        For every pair in mapping get history of every auto_rebuild tag
+        filtering all nvrs from that registry/repo pair.
+
+        :param dict reg_repo_info: map of pairs (registry, repository) to a
+            dict containing nvrs of bundle images from that repo and
+            auto_rebuild tags of that repo
+        :return: NVRs that were once tagged with an autorebuild tag
+        :rtype set(str):
+        """
+        params = {'include': 'data.brew.build'}
+        ret_nvrs = set()
+        for reg_repo_pair, info in reg_repo_info.items():
+            if not info.get('nvrs'):
+                continue
+            fltr = f"brew.build=in=({','.join(info['nvrs'])})"
+            params["filter"] = fltr
+            reg, repo = reg_repo_pair
+            for tag in info.get('auto_rebuild_tags', []):
+                tag_history = \
+                    self._pagination(f"repositories/registry/{reg}/"
+                                     f"repository/{repo}/tag/{tag}", params)
+                # If there is at least one record with nvr(brew.build) of the
+                # image, it means that image was published to a repo with
+                # auto_rebuild tag
+                if tag_history:
+                    auto_rebuild_nvrs = \
+                        [temp_tag['brew']['build'] for temp_tag in tag_history]
+                    ret_nvrs.update(auto_rebuild_nvrs)
+        return ret_nvrs
+
+    def get_auto_rebuild_taged_images(self, bundle_images):
+        """
+        Determine which bundle images are part of container repository and tag
+        that has auto-rebuilds enabled
+
+        :param list bundle_images: Images of operator bundles that should
+            be filtered
+        :return: Image NVRs that where published to repositories with auto_rebuild tag
+        :rtype: set(str)
+        """
+        reg_repo_info = {}
+        for bundle_image in bundle_images:
+            bundle_nvr = bundle_image.get('brew', {}).get('build')
+            if not bundle_nvr:
+                log.warning("One of bundle images doesn't have brew.build")
+                continue
+            if not bundle_image.get('repositories'):
+                log.warning("Bundle image %s doesn't have repositories set",
+                            bundle_nvr)
+                continue
+            # construct mapping of (registry, repository) -> {'nvrs': {bundles_nvrs}}
+            for repo in bundle_image.get('repositories'):
+                if 'registry' not in repo or 'repository' not in repo:
```

Comment:
This is safer to guard against falsy (e.g. None) values.
```suggestion
                if not (repo.get('registry') and repo.get('repository')):
```